### PR TITLE
Added health check

### PR DESCRIPTION
**Pull request by RGanor**
Add health check API that also check the authoritative api.

### CODE REVIEW

File: powerdnsadmin/routes/api.py [hunk 1]
```diff
@@ -1182,3 +1182,37 @@ def sync_domains():
...
+                                verify=Setting().get('verify_ssl_connections'))
+                                
+    except Exception as e:
+        current_app.logger.error("Health Check - Failed to query authoritative server for domain {}".format(domain_to_query.name))
+        return make_response("bad", 503)
+
+    if resp.status_code == 200:
+        return make_response("good", 200)
+    else:
+        return make_response("bad", 503)
```
Comment (jbe-dw) [inline powerdnsadmin/routes/api.py:1217]: I'm not sure if we should use 503 here, it's not PDA failing but the backend

**Review by jbe-dw — COMMENTED** [review]
I took time to give my opinion.

Regards

File: powerdnsadmin/routes/api.py [hunk 2]
```diff
@@ -1182,3 +1182,37 @@ def sync_domains():
...
+    try:
+        resp = utils.fetch_remote(urljoin(pdns_api_url, api_uri_with_prefix + api_uri),
+                                method='GET',
+                                headers=headers,
+                                accept='application/json; q=1',
+                                verify=Setting().get('verify_ssl_connections'))
+                                
+    except Exception as e:
+        current_app.logger.error("Health Check - Failed to query authoritative server for domain {}".format(domain_to_query.name))
+        return make_response("bad", 503)
```
[the remark this answers is not in the document]
Comment (jbe-dw) [reply powerdnsadmin/routes/api.py:1212]: This piece of code is already implemented in get_domain_info(). Did you try the code I posted ?

**Comment by RGanor** [comment]
@jbe-dw I have updated the code to use the domain model function and added authentication requirement.
Now, after we removed the `/ping` authentication, I think it make sense for this function to be more robust in its check, but because it is querying the pdns auth api (which can be resource intensive some time), the authentication is required to protect this endpoint.
I think that `/ping` and `/api/v1/health` should co-exist, each with different propose.

**Review by jbe-dw — CHANGES_REQUESTED** [review]
As we discussed, i'm waiting for an alternative message if there is no domain to query

**Comment by RGanor** [comment]
Hey @jbe-dw , thanks for bumping this up. I will be able fix this soon.

**Review by jbe-dw — APPROVED** [review]

**Pull request merged**